src/rviz/InteractiveMarkerViewer.cpp

```diff
@@ -49,6 +50,42 @@ FrameMarkerPtr InteractiveMarkerViewer::addFrame(
   return marker;
 }
 
+TSRMarkerPtr InteractiveMarkerViewer::visualizeTSR(
+  aikido::constraint::TSR const &_tsr,
+  int nSamples,
+  std::string basename)
+{
+  using dart::dynamics::Frame;
+  using dart::dynamics::SimpleFrame;
+  using aikido::constraint::TSR;
+
+  auto sampler = _tsr.createSampleGenerator();
+  auto state = _tsr.getSE3()->createState();
+  auto tsrMarker = std::make_shared<TSRMarker>();
+
+  if (basename == "")
+  {
+      std::ostringstream ost;
+      ost << &tsrMarker;
```

Comment:
`tsrMarker` is a `smart_ptr`, so this should be `tsrMarker.get()` to get the memory address of the heap memory. It's possible that repeated calls of this function in a loop could allocate `tsrMarker` itself at the same memory address on the stack.